Add r.rterm.preferredConsoles for R console selection from PATH

BerriJ opened the pull request:
This PR will close #1347 

It adds a new option `r.rterm.preferredConsoles` which defaults to ["R"]. You can specify you preferred R consoles like this:

<img width="712" height="241" alt="Screenshot_2026-07-20_15-06-22" src="https://github.com/user-attachments/assets/625c0d60-0c34-44ad-aadc-b751a7e47a5a" />

The extension will open the first item in that list that it finds on the users `PATH`. However, if `r.rterm.linux` is specified, it will use that path (given that this is a more precise setting, I suggest we give precedence to it).

Thank you so much for this project. I'm a daily user since many years. 

Best,
BerriJ

eitsupi commented:
Would it make more sense to make R and radian selectable via an Enum?

I recently created the arf console that can be used instead of radian, and I have personally switched from radian.
https://github.com/eitsupi/arf

BerriJ commented:
> Would it make more sense to make R and radian selectable via an Enum?
> 
> I recently created the arf console that can be used instead of radian, and I have personally switched from radian. https://github.com/eitsupi/arf

This would also work. You mean something like:

```
preference = [arf, radian, r]
```

And then this extension would search the PATH for arf, radian, and r, and use the first it finds?

This would be a very flexible solution.

grantmcdermott commented:
Fwiw, `radian` is no longer being (actively) maintained. The package readme [explicitly recommends](https://github.com/randy3k/radian?tab=readme-ov-file#important) switching over to `arf`:

> radian is no longer under active development. While I may address critical or obvious bugs, no new features will be added. If you are looking for an active alternative, I recommend checking out [arf](https://github.com/eitsupi/arf) by @eitsupi.

BerriJ commented:
@randy3k Do you have an idea how to proceed here? I could happily adjust the PR to use arf instead of radian, now that radian is not supported anymore.
I'm also happy to implement other ideas. The core feature I'd need is that vscode-R picks the specified R executable from the PATH, and we do not have to hardcode the specific path in the settings. As discussed here: https://github.com/REditorSupport/vscode-R/issues/1347

I'm also not shure if this PR is completely independent from #1684. Maybe you could tell us something about the big picture here. Many thanks in advance :)

randy3k commented:
We shouldn't reply on radian anymore (python is not really a good platform for development an R console).

I am not oppposing to the idea of "prefer arf". Though I am not sure if we want to "hard code" arf at the same time. A setting like "prefer this program if found" may be more useful?

BerriJ committed:
feat: add r.rterm.preferredConsoles for flexible R console selection

Add `r.rterm.preferredConsoles` (string array, default `["R"]`), an
ordered list of R console executable names to search for on PATH when no
explicit terminal path (e.g. `r.rterm.linux`) is set. The first entry
found on PATH is used, enabling users to prefer radian, ark, or any
other console over plain R without specifying a full path.

BerriJ commented:
> We shouldn't reply on radian anymore (python is not really a good platform for development an R console).
> 
> I am not oppposing to the idea of "prefer arf". Though I am not sure if we want to "hard code" arf at the same time. A setting like "prefer this program if found" may be more useful?

Thanks for your feedback. I agree, hardcoding R consoles is not ideal. I updated the PR and now the new setting takes a list of console names that the user prefers. So you can specify for example `["arf", "radian", "R"]` and it will use arf, if available on the PATH, otherwise it falls back to "radian" and so on. Would you mind taking a look at this?

I'll gladly adjust this further, based on your feebdack.
